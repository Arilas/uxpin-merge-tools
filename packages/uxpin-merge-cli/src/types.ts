export { ComponentImplementationInfo } from './steps/discovery/component/ComponentInfo';
export { ImplSerializationResult } from './steps/serialization/component/implementation/ImplSerializationResult';
export { serializeJSComponent } from './steps/serialization/component/implementation/javascript/serializeJSComponent';
export { serializeTSComponent } from './steps/serialization/component/implementation/typescript/serializeTSComponent';
export { DesignSystemSnapshot } from './steps/serialization/DesignSystemSnapshot';
