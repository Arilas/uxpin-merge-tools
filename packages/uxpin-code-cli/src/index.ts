export { getDesignSystemMetadata } from './getDesignSystemMetadata';
export { getDesignSystemSummary } from './getDesignSystemSummary';
