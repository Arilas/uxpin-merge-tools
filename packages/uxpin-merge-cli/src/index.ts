import 'source-map-support/register';

export { runProgram } from './program/runProgram';
export { Command } from './program/command/Command';
export { getToolVersion } from './program/utils/version/getToolVersion';

export { ComponentImplementationInfo } from './steps/discovery/component/ComponentInfo';
export { ImplSerializationResult } from './steps/serialization/component/implementation/ImplSerializationResult';
export { serializeJSComponent } from './steps/serialization/component/implementation/javascript/serializeJSComponent';
export { serializeTSComponent } from './steps/serialization/component/implementation/typescript/serializeTSComponent';
export { DesignSystemSnapshot } from './steps/serialization/DesignSystemSnapshot';
