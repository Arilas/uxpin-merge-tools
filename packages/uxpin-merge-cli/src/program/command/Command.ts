export enum Command {
  DUMP = 'dump',
  EXPERIMENT = 'experiment',
  INIT = 'init',
  CREATE_APP = 'create-app',
  GENERATE_PRESETS = 'generate-presets',
  PUSH = 'push',
  SERVER = 'server',
  SUMMARY = 'summary',
}

export const DEFAULT_COMMAND:Command = Command.EXPERIMENT;
