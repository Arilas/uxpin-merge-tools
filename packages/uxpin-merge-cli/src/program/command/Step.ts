import { DSMetadata } from '../DSMeta';

export interface Step {
  shouldRun:boolean;
  exec:(infos:DSMetadata) => any;
}
