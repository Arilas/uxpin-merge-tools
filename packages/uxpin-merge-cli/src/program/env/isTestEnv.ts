import { Environment } from './Environment';

export function isTestEnv():boolean {
  return process.env.NODE_ENV === Environment.TEST;
}
