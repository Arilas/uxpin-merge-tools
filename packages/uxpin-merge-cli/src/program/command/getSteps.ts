import { BuildOptions } from '../../steps/building/BuildOptions';
import { ProgramArgs, PushProgramArgs, ServerProgramArgs } from '../ProgramArgs';
import { getDumpCommandSteps } from './dump/getDumpCommandSteps';
import { getPushCommandSteps } from './push/getPushCommandSteps';
import { getServerCommandSteps } from './server/getServerCommandSteps';
import { Step } from './Step';
import { getSummaryCommandSteps } from './summary/getSummaryCommandSteps';

export function getSteps(args:ProgramArgs):Step[] {
  if (args.command === 'server') {
    return getServerCommandSteps(getBuildOptions(args), args);
  }

  if (args.command === 'push') {
    return getPushCommandSteps(getBuildOptions(args));
  }

  if (args.command === 'dump') {
    return getDumpCommandSteps();
  }

  if (args.command === 'summary') {
    return getSummaryCommandSteps();
  }

  return [];
}

function getBuildOptions(args:ServerProgramArgs | PushProgramArgs):BuildOptions {
  const { cwd, webpackConfig, wrapper } = args;
  return {
    projectRoot: cwd,
    webpackConfigPath: webpackConfig,
    wrapperPath: wrapper,
  };
}
