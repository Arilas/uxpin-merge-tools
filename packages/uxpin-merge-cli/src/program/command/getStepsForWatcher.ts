import { ProgramArgs } from '../args/ProgramArgs';
import { getExperimentationWatchCommandSteps } from './experimentation/getExperimentationWatchCommandSteps';
import { Step } from './Step';

export function getStepsForWatcher(args:ProgramArgs):Step[] {
  if (args.command === 'experiment') {
    return getExperimentationWatchCommandSteps(args);
  }

  return [];
}
