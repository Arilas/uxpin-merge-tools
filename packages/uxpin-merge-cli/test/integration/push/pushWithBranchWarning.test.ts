import { resolve } from 'path';
import { DirectoryResult } from 'tmp-promise';
import { Command } from '../../../src';
import { Environment } from '../../../src/program/env/Environment';
import { emptyLatestCommitStub } from '../../resources/stubs/emptyLatestCommit';
import { runUXPinMergeCommand } from '../../utils/command/runUXPinMergeCommand';
import { setTimeoutBeforeAll } from '../../utils/command/setTimeoutBeforeAll';
import { setupStubbyServer } from '../../utils/stubby/setupStubbyServer';
import { setupTempProject } from '../../utils/temp/setupTempProject';

const CURRENT_TIMEOUT:number = 60000;

setTimeoutBeforeAll(CURRENT_TIMEOUT);

describe('Push command', () => {
  describe('from branch other than master', () => {
    const sourceDir:string = resolve(__dirname, '../../resources/designSystems/twoComponentsWithConfig');
    const projectPath:string = resolve(__dirname, '../../../');
    const { getTlsPort } = setupStubbyServer(emptyLatestCommitStub);
    const { getDirectory } = setupTempProject({
      gitOptions: { branch: 'test', initialise: true },
      linkPackage: true,
      projectPath,
      sourceDir,
    });

    it('shows warning that different branches are not supported', async () => {
      // having
      const dir:DirectoryResult = getDirectory();

      // when
      // then
      const result:string = await runUXPinMergeCommand({
        cwd: dir.path,
        env: {
          UXPIN_API_DOMAIN: `0.0.0.0:${getTlsPort()}`,
          UXPIN_ENV: Environment.TEST,
        },
        params: [
          Command.PUSH,
          '--webpack-config "./webpack.config.js"',
          '--token DUMMY_TOKEN',
        ],
      });

      expect(result).toEqual(expect.stringMatching(/branch different than master are currently not supported/));
    });
  });

  describe('from master branch', () => {
    const sourceDir:string = resolve(__dirname, '../../resources/designSystems/twoComponentsWithConfig');
    const projectPath:string = resolve(__dirname, '../../../');
    const { getTlsPort } = setupStubbyServer(emptyLatestCommitStub);
    const { getDirectory } = setupTempProject({
      gitOptions: { initialise: true },
      linkPackage: true,
      projectPath,
      sourceDir,
    });

    it('does not show warning that different branches are not supported', async () => {
      // having
      const dir:DirectoryResult = getDirectory();

      // when
      // then
      const result:string = await runUXPinMergeCommand({
        cwd: dir.path,
        env: {
          UXPIN_API_DOMAIN: `0.0.0.0:${getTlsPort()}`,
          UXPIN_ENV: Environment.TEST,
        },
        params: [
          Command.PUSH,
          '--webpack-config "./webpack.config.js"',
          '--token DUMMY_TOKEN',
        ],
      });

      expect(result).not.toEqual(expect.stringMatching(/branch different than master are currently not supported/));
    });
  });
});
