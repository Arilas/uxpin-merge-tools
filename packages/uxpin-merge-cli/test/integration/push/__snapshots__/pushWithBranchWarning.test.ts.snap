// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Push command from branch other than master shows warning that different branches are not supported 1`] = `
"[33m‚ö†Ô∏è Libraries pushed from branch different than master are currently not supported by UXPin Editor!
[0m
[32m‚úÖ Library bundle uploaded successfully!
[0m[32m‚úÖ Library metadata uploaded successfully!
[0m"
`;

exports[`Push command from master branch does not show warning that different branches are not supported 1`] = `
"
[32m‚úÖ Library bundle uploaded successfully!
[0m[32m‚úÖ Library metadata uploaded successfully!
[0m"
`;
